panic: runtime error: invalid memory address or nil pointer

Terraform Version
`Terraform v0.14.5`

hashicorp/azurerm - `2.46.1`
hashicorp/null  - `3.0.0`

After running Terraform Apply  we received a panic: runtime error: invalid memory address or nil pointer dereference error.

and
```
2021-02-10T14:11:21.1456427Z [1m[31mError: [0m[0m[1mrpc error: code = Unavailable desc = transport is closing[0m
2021-02-10T14:11:21.1457111Z 
2021-02-10T14:11:21.1457800Z [0m[0m[0m
2021-02-10T14:11:21.1458474Z [31m
2021-02-10T14:11:21.1459373Z [1m[31mError: [0m[0m[1mrpc error: code = Unavailable desc = transport is closing[0m
2021-02-10T14:11:21.1459929Z 
2021-02-10T14:11:21.1460541Z [0m[0m[0m
2021-02-10T14:11:21.1462138Z [31m
2021-02-10T14:11:21.1463118Z [1m[31mError: [0m[0m[1mrpc error: code = Unavailable desc = transport is closing[0m
2021-02-10T14:11:21.1463621Z 
2021-02-10T14:11:21.1464459Z [0m[0m[0m
2021-02-10T14:11:21.1627161Z panic: runtime error: invalid memory address or nil pointer dereference
2021-02-10T14:11:21.1629850Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x3f052eb]
2021-02-10T14:11:21.1632712Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 
2021-02-10T14:11:21.1635346Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: goroutine 88 [running]:
2021-02-10T14:11:21.1639688Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/cosmos.resourceCosmosDbAccountDelete(0xc000ce4620, 0x4bb0200, 0xc0015b6300, 0x0, 0x0)
2021-02-10T14:11:21.1644772Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/cosmos/cosmosdb_account_resource.go:773 +0x24b
2021-02-10T14:11:21.1648526Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000603ef0, 0xc0001aebe0, 0xc00120d9e0, 0x4bb0200, 0xc0015b6300, 0x4bda501, 0xc001224aa8, 0xc001263500)
2021-02-10T14:11:21.1652195Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:283 +0x4a3
2021-02-10T14:11:21.1656216Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000295080, 0xc001411a10, 0xc0001aebe0, 0xc00120d9e0, 0xc001224b20, 0x0, 0x0)
2021-02-10T14:11:21.1659860Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x99
2021-02-10T14:11:21.1664528Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000f928, 0x5954b00, 0xc001216000, 0xc000ce4000, 0xc00000f928, 0xc001216000, 0xc00178ab78)
2021-02-10T14:11:21.1668079Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:885 +0x8ab
2021-02-10T14:11:21.1672016Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x501ef80, 0xc00000f928, 0x5954b00, 0xc001216000, 0xc001214000, 0x0, 0x5954b00, 0xc001216000, 0xc00120e000, 0xbb1)
2021-02-10T14:11:21.1676037Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
2021-02-10T14:11:21.1679325Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00018b800, 0x599c5c0, 0xc0000d4900, 0xc00152e200, 0xc00051d950, 0x8827600, 0x0, 0x0, 0x0)
2021-02-10T14:11:21.1682464Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1024 +0x522
2021-02-10T14:11:21.1685686Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: google.golang.org/grpc.(*Server).handleStream(0xc00018b800, 0x599c5c0, 0xc0000d4900, 0xc00152e200, 0x0)
2021-02-10T14:11:21.1688836Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1313 +0xd34
2021-02-10T14:11:21.1691783Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000c12950, 0xc00018b800, 0x599c5c0, 0xc0000d4900, 0xc00152e200)
2021-02-10T14:11:21.1695332Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:722 +0xa5
2021-02-10T14:11:21.1698502Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-02-10T14:11:21.1701426Z 2021-02-10T14:11:21.118Z [DEBUG] plugin.terraform-provider-azurerm_v2.46.1_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:720 +0xa5
2021-02-10T14:11:21.1704249Z 2021-02-10T14:11:21.121Z [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/azurerm/2.46.1/linux_amd64/terraform-provider-azurerm_v2.46.1_x5 pid=734 error="exit status 2"
2021-02-10T14:11:21.1706318Z 2021/02/10 14:11:21 [DEBUG] azurerm_subnet.pftransdatabricksprivate: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2021-02-10T14:11:21.1707996Z 2021/02/10 14:11:21 [DEBUG] azurerm_subnet.pftransdatabrickspublic: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2021-02-10T14:11:21.1709670Z 2021/02/10 14:11:21 [DEBUG] azurerm_cosmosdb_account.platform-persistent: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing

2021-02-10T14:11:21.1757770Z !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
2021-02-10T14:11:21.1758077Z 
2021-02-10T14:11:21.1758656Z Terraform crashed! This is always indicative of a bug within Terraform.
2021-02-10T14:11:21.1759395Z A crash log has been placed at "crash.log" relative to your current
2021-02-10T14:11:21.1760105Z working directory. It would be immensely helpful if you could please
2021-02-10T14:11:21.1760825Z report the crash with Terraform[1] so that we can fix this.
2021-02-10T14:11:21.1761172Z 
2021-02-10T14:11:21.1761741Z When reporting bugs, please include your terraform version. That
2021-02-10T14:11:21.1762466Z information is available on the first line of crash.log. You can also
2021-02-10T14:11:21.1763392Z get it by running 'terraform --version' on the command line.
2021-02-10T14:11:21.1763692Z 
2021-02-10T14:11:21.1764294Z SECURITY WARNING: the "crash.log" file that was created may contain 
2021-02-10T14:11:21.1765012Z sensitive information that must be redacted before it is safe to share 
2021-02-10T14:11:21.1765634Z on the issue tracker.
2021-02-10T14:11:21.1765863Z 
2021-02-10T14:11:21.1766529Z [1]: https://github.com/hashicorp/terraform/issues
2021-02-10T14:11:21.1767033Z 
2021-02-10T14:11:21.1768094Z !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
2021-02-10T14:11:21.1832289Z ##[error]Bash exited with code '1'.
2021-02-10T14:11:21.1881566Z ##[section]Finishing: Run terraform apply
```

AzureRM was upgraded from `2.41.0`  to `2.46.1`.  terraform configuration worked well with `2.41.0`

--- a/azurerm/internal/services/cosmos/cosmosdb_account_resource.go
+++ b/azurerm/internal/services/cosmos/cosmosdb_account_resource.go
@@ -768,11 +768,7 @@ func resourceCosmosDbAccountDelete(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.Name)
-	if err != nil {
-		if future.Response().StatusCode == http.StatusNoContent {
-			return nil
-		}
+	if _, err := client.Delete(ctx, id.ResourceGroup, id.Name); err != nil {
 		return fmt.Errorf("deleting CosmosDB Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
 

